src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -2325,15 +2326,43 @@ private static function getTrueInequalityAssertions(
         }
 
         if (count($notif_types) === 1) {
-            $notif_types = $notif_types[0];
+            $notif_type = $notif_types[0];
 
-            if (count($notif_types) === 1) {
-                $if_types = Algebra::negateTypes($notif_types);
+            if (count($notif_type) === 1) {
+                $if_types = Algebra::negateTypes($notif_type);
             }
         }
 
         $if_types = $if_types ? [$if_types] : [];
 
+        if ($if_types === [] && count($notif_types) === 2) {
+            foreach ($notif_types as $notif_type) {
+                foreach ($notif_type as $var => $assertions) {
+                    if (count($assertions) !== 1 || count($assertions[0]) !== 1) {
+                        $if_types = [];
+                        break 2;
+                    }
+
+                    $is_not_assertion = $assertions[0][0] instanceof IsNotType ? true : false;
+                    if (!isset($check_var)) {
```

Comment:
Please initialize your variable to null before the loop, Psalm is not great at checking conditionally initialized variable